src/Microsoft.Azure.WebJobs.Host/Loggers/FunctionInstanceLogger.cs

```diff
@@ -17,56 +17,215 @@ internal class FunctionInstanceLogger : IFunctionInstanceLogger
     {
         private readonly ILoggerFactory _loggerFactory;
 
+        private readonly ConcurrentDictionary<string, ILogger> _loggers = new ConcurrentDictionary<string, ILogger>();
+
+        private readonly ConcurrentDictionary<string, Func<ILogger, IDictionary<string, string>, bool>> _triggerLoggers = new ConcurrentDictionary<string, Func<ILogger, IDictionary<string, string>, bool>>();
+
+        private static readonly Action<ILogger, string, string, Guid, Exception> LogFunctionStarted =
+            LoggerMessage.Define<string, string, Guid>(LogLevel.Information, 0, "Executing '{Function}' (Reason='{Reason}', Id={Id})");
```

Comment:
Also... we should use an explicit `EventId` in these. We just did this in host repo (example: https://github.com/Azure/azure-functions-host/blob/3b1b0dd0a0f54098f72d6a284d485bf671a76280/src/WebJobs.Script.WebHost/Diagnostics/Extensions/ScriptHostServiceLoggerExtension.cs), and we made sure that the EventId had the correct name and also a unique id. 

You could pick some integer like 1000 to start and number them 1000, 1001, 1002, etc to keep them distinct from the numbers we have in the host repo.

Reply:
Will do.